test/server/admin/stats_handler_speed_test.cc

```diff
@@ -9,25 +9,100 @@
+};
+
+// Override the methods used by this test so that using a mock doesn't affect performance.
+class FastMockCluster : public testing::StrictMock<Upstream::MockCluster>,
+                        public testing::StrictMock<Upstream::MockPrioritySet> {
+public:
+  FastMockCluster(const std::string& name, FastMockClusterManager& cm)
+      : cm_(cm), scope_(name, cm_.store_), name_(name) {}
+
+  Upstream::ClusterInfoConstSharedPtr info() const override { return info_; }
+  const Upstream::PrioritySet& prioritySet() const override { return *this; }
+  const std::vector<Upstream::HostSetPtr>& hostSetsPerPriority() const override {
+    return host_sets_;
+  }
+
+  class ClusterInfo : public testing::StrictMock<Upstream::MockClusterInfo> {
+  public:
+    ClusterInfo(FastMockCluster& parent) : parent_(parent) {}
+
+    bool perEndpointStatsEnabled() const override { return parent_.cm_.per_endpoint_enabled_; }
+    const std::string& observabilityName() const override { return parent_.name_; }
+    Stats::Scope& statsScope() const override { return parent_.scope_; }
+
+    FastMockCluster& parent_;
+  };
+
+  FastMockClusterManager& cm_;
+  Stats::TestUtil::TestScope scope_;
+  std::shared_ptr<ClusterInfo> info_{std::make_shared<ClusterInfo>(*this)};
+  std::string name_;
+  std::vector<Upstream::HostSetPtr> host_sets_;
+};
+
+class FastMockHostSet : public testing::StrictMock<Upstream::MockHostSet> {
+public:
+  const Upstream::HostVector& hosts() const override { return hosts_; }
+
+  Upstream::HostVector hosts_;
+};
+
+class FastMockHost : public testing::StrictMock<Upstream::MockHostLight> {
+public:
+  Network::Address::InstanceConstSharedPtr address() const override { return address_; }
+  const std::string& hostname() const override { return hostname_; }
+
+  std::vector<std::pair<absl::string_view, Stats::PrimitiveCounterReference>>
+  counters() const override {
+    return host_stats_.counters();
+  }
+
+  std::vector<std::pair<absl::string_view, Stats::PrimitiveGaugeReference>>
+  gauges() const override {
+    return host_stats_.gauges();
+  }
+
+  Health coarseHealth() const override { return Upstream::Host::Health::Healthy; }
+
+  Network::Address::InstanceConstSharedPtr address_;
+  std::string hostname_;
+  mutable Upstream::HostStats host_stats_;
+};
+
 class StatsHandlerTest : public Stats::ThreadLocalRealThreadsMixin {
 public:
   StatsHandlerTest() : ThreadLocalRealThreadsMixin(1) {
     runOnMainBlocking([this]() { init(); });
     Envoy::benchmark::setCleanupHook([this]() { delete this; });
   }
 
+  static constexpr uint32_t NumClusters = 10000;
+
   void init() {
     // Benchmark will be 10k clusters each with 100 counters, with 100+
     // character names. The first counter in each scope will be given a value so
     // it will be included in 'usedonly'.
     const std::string prefix(100, 'a');
-    for (uint32_t s = 0; s < 10000; ++s) {
+    for (uint32_t s = 0; s < NumClusters; ++s) {
```

Comment:
revert this back to 1000?

Reply:
It was always 10k. I just moved it to a constant.